### ISSUE TITLE
Error 500 container_registry

### ISSUE THREAD

**Issue by stkac** [sameersbn/docker-gitlab#1688]
Hi, I get **500** error after deploy using traefik, someone know how fix it? all the other services work...
> 500
> Whoops, something went wrong on our end.

thanks in advance,

**logs of registry container:**

```
time="2018-07-30T08:47:45.723659565Z" level=info msg="redis not configured" go.version=go1.6.2 instance.id=4d3e8e3a-7d9b-4626-9dfe-f06b4ebb1402 service=registry version=v2.4.1
time="2018-07-30T08:47:45.724275748Z" level=info msg="Starting upload purge in 6m0s" go.version=go1.6.2 instance.id=4d3e8e3a-7d9b-4626-9dfe-f06b4ebb1402 service=registry version=v2.4.1
time="2018-07-30T08:47:45.765248446Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.6.2 instance.id=4d3e8e3a-7d9b-4626-9dfe-f06b4ebb1402 service=registry version=v2.4.1
time="2018-07-30T08:47:45.769591701Z" level=info msg="listening on [::]:5000, tls" go.version=go1.6.2 instance.id=4d3e8e3a-7d9b-4626-9dfe-f06b4ebb1402 service=registry version=v2.4.1

```

**my docker-compose.yml**

```
version: '3.6'

networks:
  traefik-00:
    name: traefik-00
    external: true
  lan-00:
    name: lan-00
    external: false

services:
  redis:
    restart: always
    image: sameersbn/redis
    command:
    - --loglevel warning
    volumes:
    - ../li/redis:/var/lib/redis:Z
    networks:
    - lan-00
    labels:
    - traefik.enable=false
    expose:
      - '6379'

  postgresql:
    restart: always
    image: sameersbn/postgresql:10
    volumes:
    - ../li/postgresql:/var/lib/postgresql:Z
    environment:
    - DB_USER=gitlab
    - DB_PASS=password
    - DB_NAME=gitlabhq_production
    - DB_EXTENSION=pg_trgm
    networks:
    - lan-00
    labels:
    - traefik.enable=false

  gitlab:
    restart: always
    image: sameersbn/gitlab:11.1.3
    volumes:
    - ../li/gitlab:/home/git/data:Z
    - ../li/logs:/var/log/gitlab
    - ../../../traefik/li/tls:/tls
    depends_on:
    - redis
    - postgresql
    external_links:
    - "registry:registry.demo.com"
    labels:
    - traefik.enable=true
    - traefik.backend=gitlab.demo.com
    - traefik.frontend.rule=Host:gitlab.demo.com
    - traefik.docker.network=traefik-00
    - traefik.port=80
    networks:
    - lan-00
    - traefik-00
    environment:
    - DEBUG=true

    - DB_ADAPTER=postgresql
    - DB_HOST=postgresql
    - DB_PORT=5432
    - DB_USER=gitlab
    - DB_PASS=password
    - DB_NAME=gitlabhq_production

    - REDIS_HOST=redis
    - REDIS_PORT=6379

    - GITLAB_HTTPS=true
    - SSL_SELF_SIGNED=true

    - GITLAB_HOST=gitlab.demo.com
    - GITLAB_PORT=443
    - GITLAB_SSH_PORT=10022
    - GITLAB_SECRETS_DB_KEY_BASE=secret
    - GITLAB_SECRETS_SECRET_KEY_BASE=secret
    - GITLAB_SECRETS_OTP_KEY_BASE=secret

    - GITLAB_ROOT_PASSWORD=password
    - GITLAB_ROOT_EMAIL=gitlab@demo.com


    - GITLAB_REGISTRY_ENABLED=true
    - GITLAB_REGISTRY_HOST=registry.demo.com
    - GITLAB_REGISTRY_PORT=5000
    - GITLAB_REGISTRY_API_URL=http://registry:5000
    - GITLAB_REGISTRY_CERT_PATH=/tls/cert/registry.demo.com.crt
    - GITLAB_REGISTRY_KEY_PATH=/tls/private/registry.demo.com.key
    - GITLAB_REGISTRY_ISSUER=gitlab-issuer

    - SSL_REGISTRY_KEY_PATH=/tls/private/registry.demo.com.key
    - SSL_REGISTRY_CERT_PATH=/tls/cert/registry.demo.com.crt

    - TZ=Europe/Madrid
    - GITLAB_TIMEZONE=Madrid

    - GITLAB_PAGES_ENABLED=true

    - GITLAB_BACKUP_SCHEDULE=daily
    - GITLAB_BACKUP_TIME=01:00

  registry:
    restart: always
    image: registry:2.4.1
    expose:
    - "5000"
    volumes:
    - ../li/gitlab/shared/registry:/var/lib/registry
    - ../../traefik/li/tls:/tls
    external_links:
    - "gitlab:gitlab.demo.com"
    environment:
    - VIRTUAL_HOST=registry, registry.demo.com
    - REGISTRY_LOG_LEVEL=info
    - REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY=/var/lib/registry
    - REGISTRY_AUTH_TOKEN_REALM=https://gitlab.demo.com/jwt/auth
    - REGISTRY_AUTH_TOKEN_SERVICE=container_registry
    - REGISTRY_AUTH_TOKEN_ISSUER=gitlab-issuer
    - REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE=/tls/cert/registry.demo.com.crt
    - REGISTRY_STORAGE_DELETE_ENABLED=true
    - REGISTRY_HTTP_TLS_CERTIFICATE=/tls/cert/registry.demo.com.crt
    - REGISTRY_HTTP_TLS_KEY=/tls/private/registry.demo.com.key
    - REGISTRY_HTTP_SECRET=secret
    networks:
    - traefik-00
    - lan-00
    labels:
    - traefik.enable=true
    - traefik.backend=registry.demo.com
    - traefik.frontend.rule=Host:registry.demo.com
    - traefik.docker.network=traefik-00
    - traefik.port=5000

```

**Comment by geberl** [comment]
This might be the Redis 4.0.9 problem of last week. Try fixing the version to `image: sameersbn/redis:4.0.9-1` like in https://github.com/sameersbn/docker-gitlab/blob/master/docker-compose.yml.

**Comment by stkac** [comment]
hi @geberl, yes fixed using: sameersbn/redis:4.0.9-1

thanks for your time :)
